docs/RELEASE.md

```diff
@@ -0,0 +1,126 @@
+
+## Crate Bumping
+
+Cadence: (possibly) each Merge Request. Responsible: Developer that opened the MR.
+
+Following SemVer isn't easy, but there exists [a guide](https://doc.rust-lang.org/cargo/reference/semver.html) in the Rust documentation that explains the small details on when to bump what. This process should be augmented with CI checks that utilize [`cargo-semver-checks`](https://github.com/obi1kenobi/cargo-semver-checks) and/or [`cargo-public-api`](https://github.com/Enselic/cargo-public-api). They must also pay attention to downstream dependencies that require a version bump, because they export the changed API.
+
+### Steps
+
+1. Developer opens a Merge Request with changed crates against `master`.
+2. They bump all changed crates according to SemVer.
+3. They bump all crates that export any changed types in their *public API*.
+4. They also bump all crates that inherit logic changes from relying on one of the bumped crates. 
+
+## Mainline Release
+
+Cadence: every two weeks. Responsible: Release Team.
+
+This process aims to release the `release` branch as a *Mainline* release every two weeks. It should eventually be automated.
+
+### Steps
+
+1. Check if process [Clobbering](#clobbering) needs to happen and do so, if that is the case.
+2. Check out the latest commit of `release`.
+3. Verify all CI checks of that commit.
+4. Update the `CHANGELOG.md` version and date. <!-- Here we could aggregate the PrDocs into the CHANGELOG -->
+5. Open a Merge Request against `release` for visibility.
+6. Check if there were any changes since the last release and abort, if not.
+7. Run `cargo semver-checks` and `cargo public-api` again to ensure that there are no SemVer breaks.
+8. Internal QA from the release team can happen here.
+9. Do a dry-run release to ensure that it *should* work.
+10. Merge it into `release`.
+11. Verify all CI checks.
+12. Comment that a *Mainline* release will happen from the merged commit hash.
+13. Release all changed crates to crates.io.
+14. Create a release on GitHub.
+
+## Nightly Release
+
+Cadence: every day at 00:00 UTC+1. Responsible: Release Team
+
+This process aims to release the `master` branch as a *Nightly* release. The process can start at 00:00 UTC+1 and should automatically do the following steps.
+
+1. Check out the latest commit of branch `master`.
+2. Verify all CI checks of that commit.
+3. Compare this commit to the latest `nightly*` tag and abort if there are no changes detected.
+4. Set the version of all crates to `major.0.0-nightlyYYMMDD` where `major` is the last released `major` version of that crate plus one.
+5. Tag this commit as `nightlyYYMMDD`.
+9. Do a dry-run release to ensure that it *should* work.
+7. Push this tag (the commit will not belong to any branch).
+8. Announce the intent to do a *Nightly* release from that tag in the RelEng chat.
```

Comment:
And then you assume that someone answers them at 3am? :P 

This entire process should be 100% automated. Just push the shit out there. No questions asked or whatever.